citesphere/src/test/java/edu/asu/diging/citesphere/core/service/impl/CitationManagerTest.java

```diff
@@ -37,217 +40,273 @@
     }
-    
-    @Test 
-    public void test_updateCitation_success() throws ZoteroConnectionException, CitationIsOutdatedException, ZoteroHttpStatusException {
+
+    @Test
+    public void test_updateCitation_success()
+            throws ZoteroConnectionException, CitationIsOutdatedException, ZoteroHttpStatusException {
         Mockito.when(zoteroManager.getGroupItemVersion(user, GROUP_ID, EXISTING_ID)).thenReturn(currentVersion);
         Mockito.when(customCitationRepository.mergeCitation(existingCitation)).thenReturn(existingCitation);
         ICitation updatedCitation = new Citation();
         updatedCitation.setKey(EXISTING_ID);
         updatedCitation.setVersion(new Long(2));
         Mockito.when(zoteroManager.updateCitation(user, GROUP_ID, existingCitation)).thenReturn(updatedCitation);
-        
+
         managerToTest.updateCitation(user, GROUP_ID, existingCitation);
-        Mockito.verify(citationRepository).save((Citation)updatedCitation);
+        Mockito.verify(citationRepository).save((Citation) updatedCitation);
     }
-    
-    @Test(expected=CitationIsOutdatedException.class)
-    public void test_updateCitation_conflict() throws ZoteroConnectionException, CitationIsOutdatedException, ZoteroHttpStatusException {
+
+    @Test(expected = CitationIsOutdatedException.class)
+    public void test_updateCitation_conflict()
+            throws ZoteroConnectionException, CitationIsOutdatedException, ZoteroHttpStatusException {
         Mockito.when(zoteroManager.getGroupItemVersion(user, GROUP_ID, EXISTING_ID)).thenReturn(new Long(2));
         managerToTest.updateCitation(user, GROUP_ID, existingCitation);
     }
-    
+
+    @Test
+    public void test_deleteCitation_success()
+            throws ZoteroConnectionException, GroupDoesNotExistException, ZoteroHttpStatusException {
+        /*
+         * ICitation deleteCitation = new Citation();
+         * deleteCitation.setKey(EXISTING_ID); deleteCitation.setVersion(new Long(2));
+         */
+
+        Long group3Id = new Long(GROUP_ID);
+        CitationGroup group = new CitationGroup();
+        group.setId(group3Id);
+        group.setVersion(new Long(2));
+        Optional<CitationGroup> optionalGroup = Optional.of(group);
+
+        List<ICitation> citations = new ArrayList<>();
+        citations.add(existingCitation);
+
+        List<PageRequest> requests = new ArrayList<PageRequest>();
+        PageRequest request = new PageRequest();
+        request.setCitations(citations);
+        requests.add(request);
+
+        managerToTest.deleteCitation(user, GROUP_ID, existingCitation);
+
+        Mockito.when(groupRepository.findById(new Long(GROUP_ID))).thenReturn(optionalGroup);
+        Mockito.verify(zoteroManager).deleteCitation(user, GROUP_ID, existingCitation);
+        Mockito.verify(citationRepository).delete((Citation) existingCitation);
+        Mockito.when(pageRequestRepository.findByCitations(existingCitation)).thenReturn(requests);
```

Comment:
why is that here? you already called the manager.

Reply:
also, you want to make sure that the citation has been removed from the page request citations list